File: examples/go/loop-shapes/LoopShapes.runar.go
package contract

import runar "github.com/icellan/runar/packages/runar-go"

// LoopShapes — Go port. A NON-ZERO loop start, ascending (R-102).
type LoopShapes struct {
	runar.SmartContract
	Target runar.Int `runar:"readonly"`
}

func (c *LoopShapes) Verify(seed runar.Int) {
	acc := seed
	for i := runar.Int(3); i < 7; i++ {
		acc = acc + i
	}
	runar.Assert(acc == c.Target)
}

File: examples/rust/loop-shapes/LoopShapes.runar.rs
use runar::prelude::*;

/// LoopShapes — Rust DSL port. A NON-ZERO loop start, ascending (R-102).
#[runar::contract]
struct LoopShapes {
    #[readonly]
    target: Int,
}

impl LoopShapes {
    pub fn verify(&self, seed: Int) {
        let mut acc: Int = seed;
        for i in 3..7 {
            acc = acc + i;
        }
        assert!(acc == self.target);
    }
}
